glue/viewers/profile/qt/tests/test_data_viewer.py

```diff
@@ -279,3 +279,42 @@ def test_session_back_compat(self, protocol):
         assert not viewer4.state.layers[2].visible
 
         ga.close()
+
+    def test_reset_limits(self):
+        self.viewer.add_data(self.data)
+        self.viewer.add_data(self.data2)
+        self.viewer.state.x_min = 0.2
+        self.viewer.state.x_max = 0.4
+        self.viewer.state.y_min = 0.3
+        self.viewer.state.y_max = 0.5
+        self.viewer.state.reset_limits()
+        assert self.viewer.state.x_min == 0
+        assert self.viewer.state.x_max == 4
+        assert self.viewer.state.y_min == 7
+        assert self.viewer.state.y_max == 23
+
+    def test_limits_unchanged(self):
+        # Make sure the limits don't change if a subset is created or another
+        # dataset added - they should only change if the reference data is changed
+        self.viewer.add_data(self.data)
+        self.viewer.state.x_min = 0.2
```

Comment:
I wonder if allclose check is good enough. You know floating point and all that.

Reply:
I think this is fine as it should literally return the exact same values, there is no math involved :)